libraries/chain/controller.cpp

```diff
@@ -378,7 +378,7 @@ struct controller_impl {
    }
 
    void dmlog_applied_transaction(const transaction_trace_ptr& t) {
-      if (auto dm_logger = get_deep_mind_logger()) {
+      if (auto dm_logger = get_deep_mind_logger(false)) {
```

Comment:
We do not want to log read-only or dry-run transaction traces. This likely points that we should include the transaction type in transaction_trace. The type would be useful here, but it might also be useful for other clients of transaction_trace to know if a transaction was a read-only, dry-run, etc.

Reply:
`dmlog_applied_transaction` is called in `push_scheduled_transaction` and `push_transaction`. read-only or dry-run transaction would not be a deferred transaction. In `push_transaction`, `dmlog_applied_transaction` is only called inside `if (!trx->is_transient())`. That's why I did not change `dmlog_applied_transaction`.

Reply:
Ok, I see that. You might want to add a comment that this is not called for transient transactions.

Reply:
I will add a comment to any `get_deep_mind_logger(false`).